dojo/utils.py

```diff
@@ -2108,3 +2121,100 @@ def is_scan_file_too_large(scan_file):
         if size > settings.SCAN_FILE_MAX_SIZE:
             return True
     return False
+
+
+def sla_compute_and_notify():
+    """
+    The SLA computation and notification will be disabled if the user opts out
+    of the Findings SLA on the System Settings page.
+
+    Notifications are managed the usual way, so you'd have to opt-in.
+    Exception is for JIRA issues, which would get a comment anyways.
+    """
+    def _notify(finding, title):
+        create_notification(
+            event='sla_breach',
+            title=title,
+            finding=finding,
+            sla_age=sla_age
+        )
+
+        if sla_notification_enabled:
+            if jira_config and jira_issue:
+                logger.debug("Creating JIRA comment to notify of SLA breach information.")
+                add_simple_jira_comment(jira_config, jira_issue, title)
+
+    # exit early on flags
+    if not settings.SLA_NOTIFY_ACTIVE and not settings.SLA_NOTIFY_ACTIVE_VERIFIED_ONLY:
+        logger.info("Will not notify on SLA breach per user configured flags")
+        return
+
+    jira_issue = None
+    jira_config = None
+    try:
+        system_settings = System_Settings.objects.get()
+        if system_settings.enable_finding_sla:
+            logger.info("About to process findings for SLA notifications.")
+            logger.debug("Active {}, Verified {}, Has JIRA {}, pre-breach {}".format(
+                settings.SLA_NOTIFY_ACTIVE,
+                settings.SLA_NOTIFY_ACTIVE_VERIFIED_ONLY,
+                settings.SLA_NOTIFY_WITH_JIRA_ONLY,
+                settings.SLA_NOTIFY_PRE_BREACH,
+            ))
+
+            # TODO be able to specify products
+            query = None
+            if settings.SLA_NOTIFY_ACTIVE:
+                query = Q(active=True, is_Mitigated=False, duplicate=False)
+            if settings.SLA_NOTIFY_ACTIVE_VERIFIED_ONLY:
+                query = Q(active=True, verified=True, is_Mitigated=False, duplicate=False)
+            logger.debug("My query: {}".format(query))
+
+            no_jira_findings = {}
+            if settings.SLA_NOTIFY_WITH_JIRA_ONLY:
+                logger.debug("Ignoring findings that are not linked to a JIRA issue")
+                no_jira_findings = Finding.objects.exclude(jira_issue__isnull=False)
+            findings = Finding.objects \
+                .filter(
+                    query
```

Comment:
Is it this query limited per product, or is for all active findings?

Reply:
Unless I am mistaken, it is every day at 7:30AM (TZ being defined by Django).

Reply:
All findings indeed. I think I have a TODO to eventually scope it at the Product level. But wanted to get it out first before the week-end.

Reply:
My TODO was not for that, but a JIRA scope override (JIRA SLA comment), which is now in.

The assumption here is that if you enable SLA at the System Settings level, then you implicitly want to know when you breach them -- it was kind of a missing feature IMO. So basically, just aligning with what's already in there. No plan to change that at the moment, unless one day SLA goes to the Product level instead of System Settings level.